macros/base/homing/homing_override.cfg

```diff
@@ -315,10 +315,12 @@ gcode:
         G28 Z0
 
         G91
-        {% if printer["configfile"].settings["stepper_z"]["homing_positive_dir"] == False %}
-            G0 Z{homing_zhop} F{z_drop_speed} # small Z hop to avoid grinding the bed (as we should be close to Z0 right now)
-        {% elif printer["configfile"].settings["stepper_z"]["homing_positive_dir"] == True %}
-            G0 Z-{homing_zhop} F{z_drop_speed} # small Z move in the opposite direction to avoid staying on the endstop (not dangerous since we should be at Z max)
+        {% if (printer.toolhead.position.z < homing_zhop) %}
+            # small Z hop to avoid grinding the bed (as we should be at Z min right now)
+            G0 Z{homing_zhop} F{z_drop_speed}
+        {% else %}
+            # small Z move in the opposite direction to avoid staying on the endstop (not dangerous since we should be at Z max)
+            G0 Z-{homing_zhop} F{z_drop_speed}
```

Comment:
This will not work in case some people use a homing_zhop of 0 and can lead to dangerous behavior. Using the `homing_positive_dir` is indeed necessary here.

Reply:
I can't think of any reason why anyone would do that, but I could wrap it in a check that z-hop is more than 0.

Reply:
On second thought, I just reverted this change to keep overall complexity to a minimum.